web/src/main/java/org/akaza/openclinica/control/submit/ListStudySubjectTableFactory.java

```diff
@@ -94,9 +94,34 @@
     // To avoid showing title in other pages, the request element is used to determine where the request came from.
     public TableFacade createTable(HttpServletRequest request, HttpServletResponse response) {
         locale = LocaleResolver.getLocale(request);
-        session = request.getSession();
-        TableFacade tableFacade = getTableFacadeImpl(request, response);
-        tableFacade.setStateAttr("restore");
+        session = request.getSession();      
+     
+        // new setting
+        TableFacade tableFacade = null;
+        String matchedStudyOID = null;
+        if(request.getParameter("maxRows") != null) {        	
+        	tableFacade = prepareNewTableFacade(request, response);
+        }else {
+        	// restore from the session   	        	 
+        	 TableFacade tableFacadeInSession = (TableFacade) request.getSession().getAttribute("LSStableFacade");
```

Comment:
create a static variable to store the attribute name since it is referenced in multiple places. 
E.g. private static final String LIST_SUBJECTS_TABLE_FACADE = "LSStableFacade";

Reply:
done